kernel/src/main/java/org/kframework/unparser/ToLatex.java

```diff
@@ -0,0 +1,208 @@
+            apply(new DataOutputStream(out), mod);
+            return out.toByteArray();
+        } catch (IOException e) {
+            throw KEMException.criticalError("Could not write K Module to LaTeX", e);
+        }
+    }
+
+    public static void apply(DataOutputStream out, Module mod) throws IOException {
+        writeString(out, ("\\outerModule{" + mod.name() + "}{"));
+        for (Module m: JavaConverters.seqAsJavaList(mod.imports().toSeq())) {
+            writeString(out, "\\outerImport{" + m.name() + "}");
+        }
+        writeString(out, "}{");
+        for (Sentence sent: JavaConverters.seqAsJavaList(mod.localSentences().toSeq())) {
+            apply(out, sent);
+        }
+        writeString(out, ("}{" + mod.att() + "}"));
+    }
+
+    public static byte[] apply(Sentence sent) {
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            apply(new DataOutputStream(out), sent);
+            return out.toByteArray();
+        } catch (IOException e) {
+            throw KEMException.criticalError("Could not write K term to LaTeX", e, sent);
+        }
+    }
+
+    public static void apply(DataOutputStream out, Sentence sent) throws IOException {
+        if (sent instanceof Rule) {
+            Rule rule = (Rule) sent;
+            writeString(out, "\\outerRule{");
+            apply(out, rule.body());
+            writeString(out, "}{");
+            apply(out, rule.requires());
+            writeString(out, "}{");
+            apply(out, rule.ensures());
+            writeString(out, "}{");
+            apply(out, rule.att());
+            writeString(out, "}");
+
+        } else if (sent instanceof ModuleComment) {
+            ModuleComment moduleComment = (ModuleComment) sent;
+            writeString(out, "\\outerModuleComment{" + moduleComment.comment() + "}{");
+            apply(out, moduleComment.att());
+            writeString(out, "}");
+
+        } else {
+            KEMException.criticalError("Do not know how to serialize sentence: " + sent.toString());
+        }
+    }
+
+    public static void apply(DataOutputStream out, K k) throws IOException {
+        if (k instanceof KToken) {
+            KToken tok = (KToken) k;
+
+            writeString(out, ("\\texttt{ " + tok.s() + " }"));
+
+        } else if (k instanceof KApply) {
+            KApply app = (KApply) k;
+
+            writeString(out, ("\\" + latexedKLabel(app.klabel().name())));
+
+            for (K item : app.klist().asIterable()) {
+                writeString(out, "{");
+                apply(out, item);
+                writeString(out, "}");
+            }
+
+        } else if (k instanceof KSequence) {
+            KSequence kseq = (KSequence) k;
+
+            writeString(out, "\\klabelKSequence{");
+
+            for (K item : kseq.asIterable()) {
+                apply(out, item);
+                writeString(out, "}{\\klabelKSequence{");
```

Comment:
let's call this kseq and dotk to align with kore generation.

Reply:
Done.